### PR TITLE
gitian: add FreeBSD support

### PR DESCRIPTION
**Pull request by hyc**

### CODE REVIEW

**Comment by hyc** [comment]
@TheCharlatan ping

**Comment by hyc** [comment]
The point of this last round of patches was to make the dependencies reproducible as well. Previously they weren't identical because their tarball timestamps changed with the monero revision. Now they're constant, so they should have identical hashes regardless of future monero releases.

**Comment by TheCharlatan** [comment]
Despite your boost archiver patch, I still get these errors when I build locally: https://travis-ci.org/TheCharlatan/monero/jobs/612656939?utm_medium=notification&utm_source=github_status .

File: contrib/depends/hosts/freebsd.mk [hunk 1]
```diff
@@ -0,0 +1,17 @@
+freebsd_CC=clang-8
```
Comment (TheCharlatan) [inline contrib/depends/hosts/freebsd.mk]: Does this only work for `clang-8`? Would be nice to get the requirement down to a more generic `clang` version. It also requires the system to alias its `clang` installation to `clang-8`, so probably it should just work `clang` instead.

Comment (hyc) [reply contrib/depends/hosts/freebsd.mk]: clang-8 is the same version that was actually shipped on FreeBSD 11, which is why I set that. Ubuntu offers clang 7, 8, and 9 and I didn't want to leave it ambiguous.

**Comment by hyc** [comment]
I was focused on getting the Android build dependencies 100% reproducible (which they are now, at the moment) and I might have broken something else. The freeBSD build was already 100% before my recent changes. I'm re-running all the builds now to see what's broken.

**Comment by hyc** [comment]
The android, freeBSD, and windows builds are now 100% identical. In the linux builds, the order of objects in libunwind.a still doesn't match; I've added an explicit sort command to libtool to try to fix that. A few of the native_* packages on Mac still don't match.

**Comment by hyc** [comment]
Linux dependencies are now 100% identical. Just have to look into these Mac discrepancies.

**Comment by hyc** [comment]
Fixed all of the Mac differences but one. There's a known bug in the old version of clang that the SDK uses. https://reviews.llvm.org/D50559 This bug makes the output of the Objective-C compiler completely non-deterministic. As a result, the libobjc2/arc.m file in the native_cctools package is different every time it's compiled. I don't see any solution to this with the version of clang in use.

File: contrib/depends/packages/packages.mk [hunk 2]
```diff
@@ -1,18 +1,22 @@
-ifeq ($(host_os),android)
 packages:=boost openssl zeromq libiconv
-else
-packages:=boost openssl zeromq expat ldns libiconv hidapi protobuf libusb
```
Comment (moneromooo-monero) [inline contrib/depends/packages/packages.mk]: expat and ldns have gone, was that intended ?

File: contrib/depends/patches/unwind/fix_obj_order.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
++++ config/ltmain.sh	2019-11-17 06:46:51.994402494 +0000
+@@ -7957,6 +7957,8 @@
+ 	    esac
+ 	  done
+ 	fi
++	oldobjs=`for obj in $oldobjs; do echo $obj; done | sort`
++	oldobjs=" `echo $oldobjs`"
+ 	eval cmds=\"$old_archive_cmds\"
+ 
+ 	func_len " $cmds"
```
Comment (moneromooo-monero) [inline contrib/depends/patches/unwind/fix_obj_order.patch]: Oh god don't tell me you patched libtool :|

File: contrib/depends/hosts/freebsd.mk (hunk 1 above)
Comment (TheCharlatan) [reply contrib/depends/hosts/freebsd.mk]: I'd leave it generic; it at least compiles fine on my clang 7 system.

File: contrib/depends/packages/packages.mk (hunk 2 above)
Comment (hyc) [reply contrib/depends/packages/packages.mk]: Nothing was linking to them, that I could see. No references in CMakeLists.txt anywhere either. Did I miss something?

File: contrib/depends/patches/unwind/fix_obj_order.patch (hunk 3 above)
Comment (hyc) [reply contrib/depends/patches/unwind/fix_obj_order.patch]: There was no other way to get the order of members in the libunwind.a to be deterministic.

File: contrib/depends/hosts/freebsd.mk (hunk 1 above)
Comment (hyc) [reply contrib/depends/hosts/freebsd.mk]: Given that C++ library ABIs are compiler version-dependent, I thought it was safest to use the same version FreeBSD uses. Did you run your binaries? I was able to run mine on a FreeBSD 11 VM.

File: contrib/depends/packages/packages.mk (hunk 2 above)
Comment (moneromooo-monero) [reply contrib/depends/packages/packages.mk]: If you did this on purpose, all good. I just read the diff and saw the removal. All good then.

**Comment by hyc** [comment]
When all else fails, delete the offending binary... Mac dependencies are now 100% identical.

File: contrib/depends/packages/boost.mk [hunk 4]
```diff
@@ -4,6 +4,7 @@ $(package)_download_path=https://dl.bintray.com/boostorg/release/1.64.0/source/
 $(package)_file_name=$(package)_$($(package)_version).tar.bz2
 $(package)_sha256_hash=7bcc5caace97baa948931d712ea5f37038dbb1c5d89b43ad4def4ed7cb683332
 $(package)_dependencies=libiconv
+$(package)_patches=fix_aroptions.patch
```
Comment (hyc) [inline contrib/depends/packages/boost.mk]: Just to note - this patch has now been merged into upstream boost. https://github.com/boostorg/build/pull/507

**Pull request merged**